### PR TITLE
[Open3d] new port

### PR DESCRIPTION
**Pull request by fran6co**
This is a rework of #12199

Depends on:

- https://github.com/microsoft/vcpkg/pull/17415
- https://github.com/microsoft/vcpkg/pull/17370

I disabled the GUI for now, I will enable it when I get filament working with it

### CODE REVIEW

File: ports/poissonrecon/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+  ARCHIVE DESTINATION "${INSTALL_LIB_DIR}"
+  PUBLIC_HEADER DESTINATION "${INSTALL_INCLUDE_DIR}"
+  COMPONENT dev
+)
+
+if(BUILD_TOOLS)
+    install(TARGETS PoissonReconEXE SSDRecon PointInterpolant SurfaceTrimmer EDTInHeat ImageStitching AdaptiveTreeVisualization ChunkPLY
+      DESTINATION "${INSTALL_BIN_DIR}"
+    )
+endif()
```
Comment (NancyLi1013) [inline ports/poissonrecon/CMakeLists.txt:134]: Can you add CMakeLists.txt to upstream?

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/poissonrecon/CONTROL [hunk 2]
```diff
@@ -0,0 +1,5 @@
+Source: poissonrecon
+Version: 2019-09-15
+Homepage: https://github.com/mkazhdan/PoissonRecon
+Description: Adaptive Multigrid Solvers
+Build-Depends: zlib, libpng, libjpeg-turbo
```
[the remark this answers is not in the document]
Comment (fran6co) [reply ports/poissonrecon/CONTROL:5]: I'll do the change in https://github.com/microsoft/vcpkg/pull/17370

File: ports/tinygltf/portfile.cmake [hunk 3]
```diff
@@ -8,8 +8,11 @@ vcpkg_from_github(
...
 # Put the licence file where vcpkg expects it
-file(COPY ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/tinygltf/LICENSE)
-file(RENAME ${CURRENT_PACKAGES_DIR}/share/tinygltf/LICENSE ${CURRENT_PACKAGES_DIR}/share/tinygltf/copyright)
+file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
+
+# Copy the tinygltf header files and fix the path to json
+vcpkg_replace_string(${SOURCE_PATH}/tiny_gltf.h "#include \"json.hpp\"" "#include <nlohmann/json.hpp>")
+file(INSTALL ${SOURCE_PATH}/tiny_gltf.h DESTINATION ${CURRENT_PACKAGES_DIR}/include)
+
+
```
[the remark this answers is not in the document]
Comment (fran6co) [reply ports/tinygltf/portfile.cmake:18]: I'll do the change in https://github.com/microsoft/vcpkg/pull/17415

File: ports/poissonrecon/intel-isl.patch [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+-#endif // __GNUC__ || __GNUC__ < 4
++//#endif // __GNUC__ || __GNUC__ < 4
+ 					AddAtomic( _pointWeightSum , sample.weight );
+ 				}
+ 			}
+@@ -1074,4 +1074,3 @@ std::vector< node_index_type > FEMTree< Dim , Real >::merge( FEMTree* tree )
+ 	MergeNodes( _tree , tree->_tree , map , nextIndex );
+ 	return map;
+ }
+-
```
[the remark this answers is not in the document]
Comment (fran6co) [reply ports/poissonrecon/intel-isl.patch:160]: This has been discussed in https://github.com/microsoft/vcpkg/pull/17370 already, this changes come from a fork of the project to fix the compilation. The project itself doesn't seem to that active with releases

File: ports/poissonrecon/CMakeLists.txt (hunk 1 above)
Comment (fran6co) [reply ports/poissonrecon/CMakeLists.txt:134]: There is a PR already but it didn't get merged yet

File: ports/poissonrecon/intel-isl.patch (hunk 4 above)
[the remark this answers is not in the document]
Comment (fran6co) [reply ports/poissonrecon/intel-isl.patch:160]: The fork was made by the open3d guys

[the remark this answers is not in the document]
Comment (fran6co) [reply ports/poissonrecon/intel-isl.patch:160]: Fixed in https://github.com/microsoft/vcpkg/pull/17370

File: ports/poissonrecon/CONTROL (hunk 2 above)
[the remark this answers is not in the document]
Comment (fran6co) [reply ports/poissonrecon/CONTROL:5]: Fixed in https://github.com/microsoft/vcpkg/pull/17370

File: ports/poissonrecon/CMakeLists.txt (hunk 1 above)
Comment (NancyLi1013) [reply ports/poissonrecon/CMakeLists.txt:134]: Thanks for your information. Is it this PR https://github.com/mkazhdan/PoissonRecon/pull/35 in upstream?

Comment (fran6co) [reply ports/poissonrecon/CMakeLists.txt:134]: yeah, it looks like it. But the PR is a bit all over the place

**Comment by fran6co** [comment]
> `Open3d `failed with the following errors:
> 
> ```
> 
> -- Found PythonInterp: D:/installed/x86-windows/tools/python3/python.exe (found version "3.9.2") 
> 
> CMake Error at D:/installed/x86-windows/share/pybind11/FindPythonLibsNew.cmake:210 (message):
> 
>   Python libraries not found
> 
> Call Stack (most recent call first):
> 
>   C:/a/1/s/scripts/buildsystems/vcpkg.cmake:861 (_find_package)
> 
>   D:/installed/x86-windows/share/pybind11/pybind11Tools.cmake:38 (find_package)
> 
>   D:/installed/x86-windows/share/pybind11/pybind11Common.cmake:195 (include)
> 
>   D:/installed/x86-windows/share/pybind11/pybind11Config.cmake:249 (include)
> 
>   C:/a/1/s/scripts/buildsystems/vcpkg.cmake:861 (_find_package)
> 
>   3rdparty/find_dependencies.cmake:726 (find_package)
> 
>   CMakeLists.txt:444 (include)
> 
> ```

It looks to be pybind11 not finding python. I couldn't get to build python with the latest vcpkg on windows, could that be related?

**Comment by ssheorey** [comment]
@fran6co Open3D doesn't support Python 3.9.x as yet.

**Comment by JackBoosY** [comment]
> @fran6co Open3D doesn't support Python 3.9.x as yet. Only 3.6-3.8.

If it's true, we should wait for the upstream to support newer Python.

**Comment by fran6co** [comment]
I don't think the problem is python 3.9 given that I'm disabling all python stuff. Right now it's failing on windows inside pybind11 checks that technically are not needed given the current configuration but I haven't seen an easy way to turn it off

**Comment by ssheorey** [comment]
@fran6co Please see this issue https://github.com/intel-isl/Open3D/issues/3376
Python is still needed for the build process.
It should work if we can specify the [full path to the Python executable as a CMake argument.](https://github.com/intel-isl/Open3D/issues/3376#issuecomment-833577254).

**Comment by NancyLi1013** [comment]
Can we convert this PR to draft until you have done all work? @fran6co

**Comment by fran6co** [comment]
Yes, please do. I'll come back to it when I have some time. 0.13 came out as well

**Comment by Ulysses1337** [comment]
I really appreciate your effort to take on this ill-designed behemoth. Presumably I'm not alone in my desire to just process point clouds. I don't want the bloat of an asset importer or renderer, neither do I want to inherit someone else's logger macros. Does it look like it will be feasible to make the major sideshows optional?

**Comment by fran6co** [comment]
I think it should be, my goal was to just create a baseline, get it
compiling and integrated to vcpkg with some of the features. Probably the
lifecycle it’s going to be: get all the features working first and then
making them optional later

**Comment by JackBoosY** [comment]
Any progress here?

**Comment by fran6co** [comment]
Might resume work on it in the next weeks, but no promises

**Comment by NancyLi1013** [comment]
Need to resolve the conflicts. @fran6co

**Comment by NancyLi1013** [comment]
Is work still being done for this PR? @fran6co

**Comment by fran6co** [comment]
> Is work still being done for this PR? @fran6co

No, I don’t have time to work on it

**Comment by NancyLi1013** [comment]
Thanks for your contribution and response. @fran6co 

We're closing this for now since there's been no progress for a long time. If you'd like to continue working on it, please reopen and ping us!

**Comment by hectorpiteau** [comment]
@fran6co It there a way to help you on this? I know it's been a while but it would be cool to reopen the issue

**Comment by QiuYilin** [comment]
Is there any progress on this?

**Pull request closed without merging**